addon/chart-data-updater.js

```diff
@@ -37,20 +37,36 @@ export default Ember.Object.extend({
     var chart = this.get('chart');
     var needUpdate = false;
 
-    data.forEach(function(segment, i) {
-      if (typeof chart.segments[i] !== 'undefined') {
+    data.forEach(function(segment) {
+      var currentSegment = chart.segments.findBy('label', segment.label);
+      if(chart.segments.findBy('label', segment.label)) {
```

Comment:
No, the if checkes if the data segment label does exist within the chart.segments. We could do an improvement to change the order, so `var currentSegment = chart.segments.findBy('label', segment.label);` is defined within the if block